azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/StorageService.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.azurecompute.arm.domain;
+
+import com.google.auto.value.AutoValue;
+import java.util.Date;
+import java.util.Map;
+
+import com.google.common.collect.ImmutableMap;
+import org.jclouds.javax.annotation.Nullable;
+import org.jclouds.json.SerializedNames;
+
+@AutoValue
+public abstract class StorageService {
+
+   public enum AccountType {
+
+      Standard_LRS,
+      Standard_ZRS,
+      Standard_GRS,
+      Standard_RAGRS,
+      Premium_LRS,
+      UNRECOGNIZED;
+
+      public static AccountType fromString(final String text) {
+         if (text != null) {
+            for (AccountType type : AccountType.values()) {
+               if (text.equalsIgnoreCase(type.name())) {
+                  return type;
+               }
+            }
+         }
+         return UNRECOGNIZED;
+      }
+   }
```

Comment:
This method is repeated in every enum. Consider creating a helper class with method to encapsulate this logic. Something like:
```java
@SuppressWarnings("unchecked")
public static <T extends Enum<T>> Enum<T> fromValueOrDefault(String text, Enum<T> defaultValue) {
   if (text != null) {
      EnumSet<T> elements = EnumSet.allOf(defaultValue.getClass());
      for (Enum<T> element : elements) {
         if (text.equalsIgnoreCase(element.name())) {
            return element;
         }
      }
   }
   return defaultValue;
}
```

And then just call it from all enums:
```java
public static AccountType fromValue(final String text) {
   return fromValueOrDefault(text, AccountType.UNRECOGNIZED);
}
```